supportedBackends/spark/src/test/scala/cromwell/backend/impl/spark/SparkRuntimeAttributesSpec.scala

```diff
@@ -0,0 +1,131 @@
+    }
+
+    "return an instance of itself when tries to validate a valid Number of Executors entry" in {
+      val expectedRuntimeAttributes = staticDefaults.copy(numberOfExecutors = Option(1))
+      val runtimeAttributes = createRuntimeAttributes(HelloWorld, """runtime { numberOfExecutors: 1 %s: "%s" }""").head
+      assertSparkRuntimeAttributesSuccessfulCreation(runtimeAttributes, emptyWorkflowOptions, expectedRuntimeAttributes)
+    }
+
+    "use workflow options as default if numberOfExecutors key is missing" in {
+      val runtimeAttributes = createRuntimeAttributes(HelloWorld, """runtime { %s: "%s" }""").head
+      val workflowOptions = workflowOptionsWithDefaultRA(Map(SparkRuntimeAttributes.NumberOfExecutorsKey -> JsNumber(1)))
+      assertSparkRuntimeAttributesSuccessfulCreation(runtimeAttributes, workflowOptions, staticDefaults.copy(numberOfExecutors = Some(1)))
+    }
+
+    "throw an exception when tries to validate an invalid numberOfExecutors entry" in {
+      val runtimeAttributes = createRuntimeAttributes(HelloWorld, """runtime { numberOfExecutors: "1" %s: "%s" }""").head
+      assertSparkRuntimeAttributesFailedCreation(runtimeAttributes, "Expecting numberOfExecutors runtime attribute to be an Integer")
+    }
+
+    "return an instance of itself when tries to validate a valid failOnStderr entry" in {
+      val expectedRuntimeAttributes = staticDefaults.copy(failOnStderr = true)
+      val runtimeAttributes = createRuntimeAttributes(HelloWorld, """runtime { failOnStderr: "true" %s: "%s" }""").head
+      val shouldBeIgnored = workflowOptionsWithDefaultRA(Map(FailOnStderrKey -> JsBoolean(false)))
+      assertSparkRuntimeAttributesSuccessfulCreation(runtimeAttributes, shouldBeIgnored, expectedRuntimeAttributes)
+    }
+
+    "throw an exception when tries to validate an invalid failOnStderr entry" in {
+      val runtimeAttributes = createRuntimeAttributes(HelloWorld, """runtime { failOnStderr: "yes" %s: "%s" }""").head
+      assertSparkRuntimeAttributesFailedCreation(runtimeAttributes, "Expecting failOnStderr runtime attribute to be a Boolean or a String with values of 'true' or 'false'")
+    }
+
+    "use workflow options as default if failOnStdErr key is missing" in {
+      val runtimeAttributes = createRuntimeAttributes(HelloWorld, """runtime { %s: "%s" }""").head
+      val workflowOptions = workflowOptionsWithDefaultRA(Map(FailOnStderrKey -> JsBoolean(true)))
+      assertSparkRuntimeAttributesSuccessfulCreation(runtimeAttributes, workflowOptions, staticDefaults.copy(failOnStderr = true))
+    }
+
+  }
+
+  private def buildWorkflowDescriptor(wdl: WdlSource,
+                                      inputs: Map[String, WdlValue] = Map.empty,
+                                      options: WorkflowOptions = WorkflowOptions(JsObject(Map.empty[String, JsValue])),
+                                      runtime: String = "") = {
```

Comment:
It would, the default value will be removed to ensure that the method is called with the proper arguments